File: sast-python3/case/completeness/single_app_tracing/expression/special_expression/spread_operator_008_F.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = false
# evaluation item = 完整度->单应用跟踪完整度->表达式->语言特有的表达式
# scene introduction = 扩展运算符
# level = 2
# bind_url = completeness/single_app_tracing/expression/special_expression/spread_operator_008_F
# evaluation information end
import os


def spread_operator_008_F(taint_src):
    collectArgs(['prefix', taint_src, 'suffix'])


def collectArgs(*args):
    taint_sink('_')


def taint_sink(o):
    os.system(o)


if __name__ == '__main__':
    taint_src = "taint_src_value"
    spread_operator_008_F(taint_src)

File: sast-js/case/accuracy/field_sensitive/one_dimensional_collection/numeric_index_state_no_solver/spread_operator_008_F.js
// evaluation information start
// real case = true
// evaluation item = 准确度->对象敏感与域敏感分析->区分一维字典/列表/数组的不同元素->索引值为数字的场景，能够区分不同索引上特定元素的状态(无需求解)
// scene introduction = 数组索引->扩展运算符
// level = 3
// bind_url = accuracy/field_sensitive/one_dimensional_collection/numeric_index_state_no_solver/spread_operator_007_T
// evaluation information end

function spread_operator_007_T(__taint_src) {
  const foo = __taint_src;
  const bar = "hello world";
  const [r1, r2, ...rest] = [123, 456, foo, bar];
  __taint_sink(rest[1]);
}

function __taint_sink(o) {}
